### PR TITLE
Add latam spanish keyboard layout

### PR DESCRIPTION
**Pull request by JuanoD**
Latin america uses a different keyboard from Spain. It makes it really hard to configure a keyboard.

## Description

Added spanish (latam) layout.
Changed keys font to monospace to handle keys with AltGr modifiers correctly.

image
image

### CODE REVIEW

**Review by fauxpark — CHANGES_REQUESTED** [review]
The corresponding header file does not exist in the qmk_firmware repository. You need to open a PR there first.

**Comment by JuanoD** [comment]
Here is the PR to qmk_firmware https://github.com/qmk/qmk_firmware/pull/22542

**Comment by JuanoD** [comment]
Is there anything stopping this from being merged?
My commit for the firmware headers is already merged into main.
If any other changes are needed, let me know.

**Comment by yanfali** [comment]
This looks pretty good, but I'm hardly an expert. @precondition any objections?

File: src/i18n/keymap_extras/keymap_spanish_latin_america.js [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+  // Row 1
+  'ALGR(KC_GRV)': { name: '¬', title: 'ES_NOT' },
+  'ALGR(KC_EQL)': { name: '\\', title: 'ES_BSLS' },
+  // Row 2
+  'ALGR(KC_Q)': { name: '@', title: 'ES_AT' },
+  'ALGR(KC_E)': { name: '€', title: 'ES_EUR' },
+  'ALGR(KC_RBRC)': { name: '~', title: 'ES_TILD' },
+  // Row 3
+  'ALGR(KC_QUOT)': { name: '^', title: 'ES_CIRC' },
+  'ALGR(KC_NUHS)': { name: '`', title: 'ES_ACUT' },
```
Comment (precondition) [inline src/i18n/keymap_extras/keymap_spanish_latin_america.js:163]:
```suggestion
  'ALGR(KC_NUHS)': { name: '`', title: 'ES_GRV (dead)' },
```

WARNING: `ES_GRV` would be the canonical alias for a dead grave key but it was forgotten in https://github.com/qmk/qmk_firmware/pull/22542 — [see how there is nothing after the declaration of `ES_CIRC`](https://github.com/JuanoD/qmk_firmware/blob/9aa4049d5603c67e7d4db564d25d1b4eab35c235/data/constants/keycodes/extras/keycodes_spanish_latin_america_0.0.1.hjson#L335) — so `ES_GRV` technically does not exist yet in qmk_firmware, and should be added.

**Review by precondition — CHANGES_REQUESTED** [review]
Comparing the output of `node qmk_configurator/src/i18n/keymap_extras/convert_keymap_extras_header.js qmk_firmware/quantum/keymap_extras/keymap_spanish_latin_america.h` and the suggested file `qmk_configurator/src/i18n/keymap_extras/keymap_spanish_latin_america.js` unearthed multiple inconsistencies between what is said in the keymap header in QMK firmware repository and what is shown here.

I'm getting the sense you manually edited the pre-existing file for the Spanish host OS layout, but the intended, albeit undocumented, workflow is to run the `convert_keymap_extras_header.js` on a keymap_extras header file from qmk_firmware.

**Review by precondition — APPROVED** [review]

**Comment by yanfali** [comment]
Thank you so much @precondition I have no objections to this being merged. @noroadsleft please merge if you have no objections

**Review by yanfali — APPROVED** [review]
Good to go

**Comment by yanfali** [comment]
@fauxpark either dismiss your review or approve. Thanks

**Review by fauxpark — APPROVED** [review]

**Review by noroadsleft — CHANGES_REQUESTED** [review]
Just one thing...

File: src/i18n/keymap_extras/keymap_spanish_latin_america.js [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+  // Row 2
+  'S(KC_LBRC)': { name: '¨', title: 'ES_DIAE (dead)' },
+  KC_LCBR: { name: '¨', title: 'ES_DIAE (dead)' },
+  'S(KC_RBRC)': { name: '*', title: 'ES_ASTR' },
+  KC_RCBR: { name: '*', title: 'ES_ASTR' },
+  // Row 3
+  'S(KC_QUOT)': { name: '[', title: 'ES_LBRC' },
+  KC_DQUO: { name: '[', title: 'ES_LBRC' },
+  // Row 4
+  'S(KC_NUHS)': { name: ']', title: 'ES_RBCR' },
```
[the remark this answers is not in the document]
Comment (JuanoD) [reply src/i18n/keymap_extras/keymap_spanish_latin_america.js]: Ready

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
¡Gracias por tu contribución! :smile:

**Pull request merged**